ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -75,9 +76,22 @@ export default Route.extend({
 
   beforeModel({ to: { queryParams } }) {
     const secret = this.secretParam();
+    const secretEngine = this.modelFor('vault.cluster.secrets.backend');
     return this.buildModel(secret, queryParams).then(() => {
       const parentKey = parentKeyForKey(secret);
       const mode = this.routeName.split('.').pop();
+      // for kv v2, redirect users from the old url to the new engine url (1.15.0 +)
+      if (isAddonEngine(secretEngine.type, secretEngine.version)) {
+        // if no secret param redirect to the create route
+        // if secret param they are either viewing or editing secret so navigate to the details route
+        return !secret
+          ? this.router.transitionTo('vault.cluster.secrets.backend.kv.create', secretEngine.id)
+          : this.router.transitionTo(
+              'vault.cluster.secrets.backend.kv.secret.details',
+              secretEngine.id,
+              secret
+            );
+      }
```

Comment:
Just curious if there is a case where this route is accessed and the engine type is not `kv` since the `isAddonEngine` would return `true` for a few other different types. PKI came to mind but I think that's been converted to fully use the engine? It might be worth considering explicitly looking for `secretEngine.type === 'kv' && secretEngine.version === 2` given the redirect routes.